Re-add relationship constraint documentation (#490)
neo4j/docs-cypher: modules/ROOT/pages/constraints/index.adoc

* Create new relationships without all the properties or where the combination of property values is not unique.
* Remove one of the mandatory properties.
* Update the properties so that the combination of property values is no longer unique.

[NOTE]
====

> Node key constraints, relationship key constraints, node property existence constraints, and relationship property existence constraints are only available in Neo4j Enterprise Edition.

Reviewer: ```suggestion
Node key constraints, relationship key constraints, node property existence constraints, and relationship property existence constraints are not available in Neo4j Community Edition.
```

They are available on all Aura environments, as far as I know.

Author: I don't know, that part of the sentence was there before and I didn't change it (only added the new relationship key to the list)
